compiler/rustc_middle/src/mir/syntax.rs

```diff
@@ -429,6 +451,9 @@ pub enum NonDivergingIntrinsic<'tcx> {
     /// If the argument is `false`, this operation is equivalent to `TerminatorKind::Unreachable`.
     Assume(Operand<'tcx>),
 
+    // Denotes a call to the intrinsic functions `likely`, `unlikely` and `unpredictable`.
+    Expect(Operand<'tcx>, ExpectKind),
```

Comment:
For MIR opts:
- SimplifyLocals: should this statement be removed when `Operand` is an unused local?
- DeadStoreElimination: when the operand is a local that is never read again?
- SimplifyConstCondition: when the operand if a constant?

Reply:
Opt passes updated